File: src/CombatAnalysis.CutomerBL/Services/RequestToConnectService.cs
using AutoMapper;
using CombatAnalysis.CustomerBL.DTO;
using CombatAnalysis.CustomerBL.Interfaces;
using CombatAnalysis.CustomerDAL.Entities;
using CombatAnalysis.CustomerDAL.Interfaces;

namespace CombatAnalysis.CustomerBL.Services;

internal class RequestToConnectService : IService<RequestToConnectDto, int>
{
    private readonly IGenericRepository<RequestToConnect, int> _repository;
    private readonly IMapper _mapper;

    public RequestToConnectService(IGenericRepository<RequestToConnect, int> repository, IMapper mapper)
    {
        _repository = repository;
        _mapper = mapper;
    }

    public Task<RequestToConnectDto> CreateAsync(RequestToConnectDto item)
    {
        if (item == null)
        {
            throw new ArgumentNullException(nameof(RequestToConnectDto), $"The {nameof(RequestToConnectDto)} can't be null");
        }

        return CreateInternalAsync(item);
    }

    public async Task<int> DeleteAsync(int id)
    {
        var rowsAffected = await _repository.DeleteAsync(id);

        return rowsAffected;
    }

    public async Task<IEnumerable<RequestToConnectDto>> GetAllAsync()
    {
        var allData = await _repository.GetAllAsync();
        var result = _mapper.Map<List<RequestToConnectDto>>(allData);

        return result;
    }

    public async Task<RequestToConnectDto> GetByIdAsync(int id)
    {
        var result = await _repository.GetByIdAsync(id);
        var resultMap = _mapper.Map<RequestToConnectDto>(result);

        return resultMap;
    }

    public async Task<IEnumerable<RequestToConnectDto>> GetByParamAsync(string paramName, object value)
    {
        var result = await Task.Run(() => _repository.GetByParam(paramName, value));
        var resultMap = _mapper.Map<IEnumerable<RequestToConnectDto>>(result);

        return resultMap;
    }

    public Task<int> UpdateAsync(RequestToConnectDto item)
    {
        if (item == null)
        {
            throw new ArgumentNullException(nameof(RequestToConnectDto), $"The {nameof(RequestToConnectDto)} can't be null");
        }

        return UpdateInternalAsync(item);
    }


    private async Task<RequestToConnectDto> CreateInternalAsync(RequestToConnectDto item)
    {
        var map = _mapper.Map<RequestToConnect>(item);
        var createdItem = await _repository.CreateAsync(map);
        var resultMap = _mapper.Map<RequestToConnectDto>(createdItem);

        return resultMap;
    }

    private async Task<int> UpdateInternalAsync(RequestToConnectDto item)
    {
        var map = _mapper.Map<RequestToConnect>(item);
        var rowsAffected = await _repository.UpdateAsync(map);

        return rowsAffected;
    }
}

File: src/CombatAnalysis.WebApp/ClientApp/src/services/RequestToConnectService.js
import HttpClient from '../helpers/HttpClient';

class RequestToConnectService {
    _httpClient = null;

    constructor() {
        this._httpClient = new HttpClient();
    }

    async getAllAsync() {
        try {
            const result = await this._httpClient.getAsync("RequestToConnect");
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async getByIdAsync(id) {
        try {
            const result = await this._httpClient.getAsync(`RequestToConnect/${id}`);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async searchByToUserIdAsync(id) {
        try {
            const result = await this._httpClient.getAsync(`RequestToConnect/searchByToUserId/${id}`);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async searchByOwnerIdAsync(id) {
        try {
            const result = await this._httpClient.getAsync(`RequestToConnect/searchByOwnerId/${id}`);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async createAsync(data) {
        try {
            const result = await this._httpClient.postAsync("RequestToConnect", data);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async updateAsync(data) {
        try {
            const result = await this._httpClient.putAsync("RequestToConnect", data);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async deleteAsync(id) {
        try {
            const result = await this._httpClient.deleteAsync(`RequestToConnect/${id}`);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }
}

export default RequestToConnectService;